addons/web/views/webclient_templates.xml

```diff
@@ -60,7 +60,7 @@
                 <main>
                     <t t-out="0"/>
                 </main>
-                <footer t-if="not no_footer" id="bottom" data-anchor="true" t-attf-class="bg-light o_footer">
+                <footer t-cache="(website, no_footer, no_copyright)" t-if="not no_footer" id="bottom" data-anchor="true" t-attf-class="bg-light o_footer">
```

Comment:
That would definitely be a concern IMHO.

Having module B and C depending of module A but not of each other and that would want to add a cache key seems legit.

Eg `website_blog` and `website_sale` could both want to add something in a `t-cache` of website, right?

Reply:
@rdeodoo It turns out that without the parentheses, in python it's still a tuple, so the inheritance technique with add works.

Reply:
That's incredibly error-prone. You can't make an API that allows developers to put a tuple of free-form Python expressions, yet at the same time expect them to provide it in a completely inflexible manner. 

Either we are honest and we don't claim any modular support in this directive (because why is it different from `t-if`?). Or we implement something different to support modularity in a real manner, like we do for `class` attributes.

From the look of it the first option is fine, each module is responsible for its own additions.